# Palabras reservadas

Existen ciertas palabras que tienen significado especial para el intérprete de Python. Estas no pueden utilizarse para ningún otro fin (como ser nombrar valores) excepto para el que han sido creadas. Estas son:

* and.
* as.
* assert.
* break.
* class.
* continue.
* def.
* del.
* elif.
* else.
* except.
* exec.
* finally.
* for.
* from.
* global.
* if.
* import.
* in.
* is.
* lambda.
* not.
* or.
* pass.
* print.
* raise.
* return.
* try.
* while.
* with.
* yield.

# Reglas y convención de nombres

Algunas reglas y convenciones de nombres para las variables y constantes:

Nunca use símbolos especiales como !, @, #, $, %, etc.

El primer carácter no puede ser un número o dígito.

Las constantes son colocadas dentro de módulos Python y significa que no puede ser cambiadas.

Los nombres de constante y variable debería tener la combinación de letras en minúsculas (de a a la z) o MAYÚSCULAS (de la A a la Z) o dígitos (del 0 al 9) o un underscore (_). Por ejemplo:

 * snake_case
 * MACRO_CASE
 * camelCase
 * CapWords

Los nombres que comienzan con guión bajo (simple _ o doble __) se reservan para variables con significado especial

No pueden usarse como identificadores, las palabras reservadas .

# Sentencia del

La sentencia **del** se define recursivamente muy similar a la forma en el cual se define la asignación. A continuación unos ejemplos donde se inicializan variables:

In [1]:
cadena, numero, lista = "Hola MET", 123456, [7,8,9,0]

In [2]:
tupla = (11, "Chau MET", True, None)

In [3]:
diccionario = {"nombre":"Rodrigo","apellido":"Rodriguez"}

Luego de inicializar las variables del código anterior, usted puede usar la función vars() para obtener un diccionario conteniendo ámbito actual de las variables, ejecutando:

In [4]:
vars()

{'__name__': '__main__',
 '__doc__': 'Automatically created module for IPython interactive environment',
 '__package__': None,
 '__loader__': None,
 '__spec__': None,
 '__builtin__': <module 'builtins' (built-in)>,
 '__builtins__': <module 'builtins' (built-in)>,
 '_ih': ['',
  'cadena, numero, lista = "Hola MET", 123456, [7,8,9,0]',
  'tupla = (11, "Chau MET", True, None)',
  'diccionario = {"nombre":"Rodrigo","apellido":"Rodriguez"}',
  'vars()'],
 '_oh': {},
 '_dh': ['/Users/andres/projects/python-met/clase03'],
 'In': ['',
  'cadena, numero, lista = "Hola MET", 123456, [7,8,9,0]',
  'tupla = (11, "Chau MET", True, None)',
  'diccionario = {"nombre":"Rodrigo","apellido":"Rodriguez"}',
  'vars()'],
 'Out': {},
 'get_ipython': <bound method InteractiveShell.get_ipython of <ipykernel.zmqshell.ZMQInteractiveShell object at 0x1114b09e8>>,
 'exit': <IPython.core.autocall.ZMQExitAutocall at 0x11150cb38>,
 'quit': <IPython.core.autocall.ZMQExitAutocall at 0x11150cb38>,
 '_': '',
 '__': '',


Si desea eliminar la referencia a la variable cadena, ejecuta:

In [5]:
del cadena

In [6]:
vars()

{'__name__': '__main__',
 '__doc__': 'Automatically created module for IPython interactive environment',
 '__package__': None,
 '__loader__': None,
 '__spec__': None,
 '__builtin__': <module 'builtins' (built-in)>,
 '__builtins__': <module 'builtins' (built-in)>,
 '_ih': ['',
  'cadena, numero, lista = "Hola MET", 123456, [7,8,9,0]',
  'tupla = (11, "Chau MET", True, None)',
  'diccionario = {"nombre":"Rodrigo","apellido":"Rodriguez"}',
  'vars()',
  'del cadena',
  'vars()'],
 '_oh': {4: {...}},
 '_dh': ['/Users/andres/projects/python-met/clase03'],
 'In': ['',
  'cadena, numero, lista = "Hola MET", 123456, [7,8,9,0]',
  'tupla = (11, "Chau MET", True, None)',
  'diccionario = {"nombre":"Rodrigo","apellido":"Rodriguez"}',
  'vars()',
  'del cadena',
  'vars()'],
 'Out': {4: {...}},
 'get_ipython': <bound method InteractiveShell.get_ipython of <ipykernel.zmqshell.ZMQInteractiveShell object at 0x1114b09e8>>,
 'exit': <IPython.core.autocall.ZMQExitAutocall at 0x11150cb38>,
 'quit': <IPyt

Como pudo ver en el ejemplo anterior que elimino la referencia a la variable cadena, incluso al volver a la función vars() ya no sale en el ámbito de variables disponibles.

La eliminación de una lista de objetivos elimina recursivamente cada objetivo, de izquierda a derecha.

In [7]:
del numero, lista, tupla, diccionario

In [8]:
vars()

{'__name__': '__main__',
 '__doc__': 'Automatically created module for IPython interactive environment',
 '__package__': None,
 '__loader__': None,
 '__spec__': None,
 '__builtin__': <module 'builtins' (built-in)>,
 '__builtins__': <module 'builtins' (built-in)>,
 '_ih': ['',
  'cadena, numero, lista = "Hola MET", 123456, [7,8,9,0]',
  'tupla = (11, "Chau MET", True, None)',
  'diccionario = {"nombre":"Rodrigo","apellido":"Rodriguez"}',
  'vars()',
  'del cadena',
  'vars()',
  'del numero, lista, tupla, diccionario',
  'vars()'],
 '_oh': {4: {...}, 6: {...}},
 '_dh': ['/Users/andres/projects/python-met/clase03'],
 'In': ['',
  'cadena, numero, lista = "Hola MET", 123456, [7,8,9,0]',
  'tupla = (11, "Chau MET", True, None)',
  'diccionario = {"nombre":"Rodrigo","apellido":"Rodriguez"}',
  'vars()',
  'del cadena',
  'vars()',
  'del numero, lista, tupla, diccionario',
  'vars()'],
 'Out': {4: {...}, 6: {...}},
 'get_ipython': <bound method InteractiveShell.get_ipython of <ipykernel.zmq

# Sentencia global

La sentencia global es una declaración que se mantiene para todo el bloque de código actual. Eso significa que los identificadores listados son interpretados como globales. Eso podría ser imposible asignar a una variable global sin la sentencia global, aunque las variables libres pueden referirse a globales sin ser declaradas globales.

In [9]:
variable1 = "variable original"

In [10]:
def modificar_variable_global_1():
    variable1 = "variable global modificada"

In [11]:
modificar_variable_global_1()

In [ ]:
# que valor tiene variable1????

Probemos usando global

In [12]:
def modificar_variable_global_2():
    global variable1
    variable1 = "variable global modificada"

In [13]:
modificar_variable_global_2()

In [16]:
# ahora si funciona

# Operadores de asignaciones

Los operadores de asignación se utilizan para

Existe en Python todo un grupo de operadores los cuales le permiten básicamente asignar un valor a una variable, usando el operador “=”. Con estos operadores pueden aplicar la técnica denominada asignación aumentada.

# Operador =

El operador igual a, (=), es el más simple de todos y asigna a la variable del lado izquierdo cualquier variable o resultado del lado derecho.

# Operador +=

El operador += suma a la variable del lado izquierdo el valor del lado derecho.

In [17]:
r = 5; r += 10; r

15

En el ejemplo anterior si la variable “r” es igual a 5 y r += 10, entonces la variable “r” sera igual a 15. Su equivalente seria el siguiente:

In [18]:
r = 5; r = r + 10; r

15

# Operador -=

El operador -= resta a la variable del lado izquierdo el valor del lado derecho.

In [19]:
r = 5; r -= 10; r

-5

En el ejemplo anterior si la variable “r” es igual a 5 y r -= 10, entonces la variable “r” sera igual a -5. Su equivalente seria el siguiente:

In [20]:
r = 5; r = r - 10; r

-5

#  Operador *=

El operador *= multiplica a la variable del lado izquierdo el valor del lado derecho.

In [21]:
r = 5; r *= 10; r

50

En el ejemplo anterior si la variable “r” es igual a 5 y r *= 10, entonces la variable “r” sera igual a 50. Su equivalente seria el siguiente:

In [22]:
r = 5; r = r * 10; r

50

# Operador //=

El operador //= calcula la división entera a la variable del lado izquierdo el valor del lado derecho.

In [26]:
r = 5; r //= 10; r

0

En el ejemplo anterior si la variable “r” es igual a 5 y r //= 10, entonces la variable “r” sera igual a 0. Su equivalente seria el siguiente:

In [27]:
r = 5; r = r // 10; r

0

# Operador **=

El operador **= calcula el exponente a la variable del lado izquierdo el valor del lado derecho.

In [28]:
r = 5; r **= 10; r

9765625

En el ejemplo anterior si la variable “r” es igual a 5 y r **= 10, entonces la variable “r” sera igual a 9765625. Su equivalente seria el siguiente:

In [29]:
r = 5; r = r ** 10; r

9765625

# Operador /=

El operador /= divide a la variable del lado izquierdo el valor del lado derecho.

In [31]:
r = 5; r /= 10; r

0.5

En el ejemplo anterior si la variable “r” es igual a 5 y r /= 10, entonces la variable “r” sera igual a 0.5 . Su equivalente seria el siguiente:

In [32]:
r = 5; r = r / 10; r

0.5

# Operador %=

El operador %= devuelve el resto de la división a la variable del lado izquierdo el valor del lado derecho.

In [33]:
r = 5; r %= 10; r

5

En el ejemplo anterior si la variable “r” es igual a 5 y r %= 10, entonces la variable “r” sera igual a 5. Su equivalente seria el siguiente:

In [34]:
r = 5; r = r % 10; r

5

# Asignación aumentada

Es frecuente que una variable tenga que ser definida de nuevo en función de sí misma. Normalmente usted escribir la siguiente sintaxis:

In [36]:
contador = 1

In [37]:
contador = contador + 1

El código anterior, se puede abreviar a su equivalente, usando la asignación aumentada, de la siguiente manera:

In [39]:
contador += 1

El código anterior, no sólo es más corto de escribir, sino también más eficiente en tiempo de ejecución.